### ISSUE TITLE
[BUG]: time series test transformation not working as expected

### ISSUE THREAD

**Issue by mfahadakbar** [pycaret/pycaret#3301]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

I am using transform_target='box-cox' in the setup. 
transformation applied to y_train seems fine, but the transformation applied to the test does not make sense.
to verify, I took the y_train_transformed and y_test_transformed from pycaret. Then outside pycaret, I applied sktime boxcox transformation. 

transformation on the y_train are exactly the same, however, test transformations are different. Here is the link to the notebook

https://colab.research.google.com/drive/1tGy-K0RnqNmFH-vkjBA3xRrgcLxoAPRH?usp=sharing

### Reproducible Example

```python
from pycaret.time_series import TSForecastingExperiment
exp = TSForecastingExperiment()
exp.setup(data=df, index='Date', fh=[1,2], session_id=42,transform_target='box-cox')

# box cox with sktime
from sktime.transformations.series.boxcox import BoxCoxTransformer
bc = BoxCoxTransformer()
bc.fit_transform(exp.y_train.to_numpy().reshape(-1, 1))

# compare transformation
import pandas as pd
df = pd.DataFrame(bc.transform(exp.y_test.to_numpy().reshape(-1, 1)),exp.y_test_transformed).reset_index()
df.columns = ['sktime_test_transformed','pycaret_test_transformed']
df['y_test'] = exp.y_test.values
df
```

### Expected Behavior

transformation on test data produced by pycaret and sktime should be the same

### Actual Results

```python-traceback
sktime_test_transformed	pycaret_test_transformed	y_test
0	42.674049	53.379534	403
1	37.583487	46.517722	320
```

### Installed Versions

<details>
System:
    python: 3.8.10 (default, Nov 14 2022, 12:59:47)  [GCC 9.4.0]
executable: /usr/bin/python3
   machine: Linux-5.10.147+-x86_64-with-glibc2.29

PyCaret required dependencies:
                 pip: 22.0.4
          setuptools: 57.4.0
             pycaret: 3.0.0rc8
             IPython: 7.9.0
          ipywidgets: 7.7.1
                tqdm: 4.64.1
               numpy: 1.21.6
              pandas: 1.3.5
              jinja2: 2.11.3
               scipy: 1.7.3
              joblib: 1.2.0
             sklearn: 1.0.2
                pyod: 1.0.7
            imblearn: 0.8.1
   category_encoders: 2.6.0
            lightgbm: 3.3.5
               numba: 0.56.4
            requests: 2.28.2
          matplotlib: 3.6.3
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.5.0
             kaleido: 0.2.1
         statsmodels: 0.13.5
              sktime: 0.16.0
               tbats: 1.1.2
            pmdarima: 2.0.2
              psutil: 5.9.4

PyCaret optional dependencies:
                shap: Not installed
           interpret: Not installed
                umap: Not installed
    pandas_profiling: 1.4.1
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
             xgboost: 0.90
            catboost: Not installed
              kmodes: Not installed
             mlxtend: 0.14.0
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: Not installed
            hyperopt: 0.1.2
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
                nltk: 3.7
            pyLDAvis: Not installed
              gensim: 3.6.0
               spacy: 3.4.4
           wordcloud: 1.8.2.2
            textblob: 0.15.3
               fugue: Not installed
           streamlit: Not installed
             prophet: 1.1.2

</details>

**Comment by ngupta23** [comment]
Hi @mfahadakbar This is by design. The idea is that when looking at test data, I should be able to use any data that came before that. Hence for test data, we fit using the entire data (train which comes before test + test data) and then just take the test indices based on the transformed data. As you can see, the highlighted values match the transformed dataset. 

![image](https://user-images.githubusercontent.com/33585645/217689388-230ad7af-9b45-4037-a07e-6dddde752356.png)

**Comment by mfahadakbar** [comment]
Hi Nikhil,

In my understanding, test data is aken to future data, to which we have no
access to. So 'fitting' should not be applied to any sort of test data. Say
, in my test data there are some extreme values. Now If we fit a Z scaler
on test+train , the fit will capture the mean of the values, and the mean
is effected by the extreme value in the test data.. that mean will then be
used to calculate the transformed values of the entire feature... This
implies a leakage, we captured the impact of variable/label that was
supposed to be hidden...

Or am I missing something here?

Thanks

**Comment by ngupta23** [comment]
Hi Fahad,

I want to clarify, we are not fitting the test data. When we do cross-validation training, the test data is entirely unseen. But for any metric evaluations after training, in order to compute any transformations on test data, we can use any data up to that point, including the training portion since that would already be known at the time of the test. This would not be considered leakage.

Hope that clarifies.

**Comment by ngupta23** [comment]
@mfahadakbar Closing this out. Feel free to reopen if you have any more questions.
